### PR TITLE
robin-hood-hashing: add package_type (and generate v2 packages)

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I've dropped versions < 3.11.3, they are missing an include of limits and I don't want to waste time patching them.

File: recipes/robin-hood-hashing/config.yml [hunk 1]
```diff
@@ -1,15 +1,7 @@
 versions:
-  "3.7.0":
```
Comment (prince-chrismc) [inline recipes/robin-hood-hashing/config.yml:2]: Why drop the older versions? 

I liked you PRs with summaries:)

Comment (SpaceIm) [reply recipes/robin-hood-hashing/config.yml:2]: explained above ;) https://github.com/conan-io/conan-center-index/pull/16617#issuecomment-1475312013

Comment (RubenRBS) [reply recipes/robin-hood-hashing/config.yml:2]: Can you take a look at https://github.com/SpaceIm/conan-center-index/pull/15? merge it if it looks ok to you and the rest of the PR lgtm :)

Comment (prince-chrismc) [reply recipes/robin-hood-hashing/config.yml:2]: > explained above ;) [#16617 (comment)](https://github.com/conan-io/conan-center-index/pull/16617#issuecomment-1475312013)

Derp thank you I absolutely missed that :)

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**